wolfssl/openssl/compat_types.h

```diff
@@ -0,0 +1,39 @@
+/* compat_types.h
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+/*
+ * Move types that cause cyclical dependency errors here.
+ */
+
+#ifndef WOLFSSL_OPENSSL_COMPAT_TYPES_H_
+#define WOLFSSL_OPENSSL_COMPAT_TYPES_H_
+
+#include <wolfssl/wolfcrypt/hmac.h>
```

Comment:
Although this works a better reference include pattern is:

```
#include <wolfssl/wolfcrypt/settings.h>
#include <wolfssl/wolfcrypt/types.h>
#include <wolfssl/wolfcrypt/hmac.h>
```

Reply:
Added the includes. This is indeed how all of our includes should look like.